### DISCUSSION TITLE
Pyinstaller with Kivy & PyQt5 GUIs?

### DISCUSSION THREAD

**Discussion by EricTomaszewski** [pyinstaller/pyinstaller#8052] [Help] [answered]
I've built 2 simple applications (input, click button, output based on input) with both: Kivy & PyQt5 to see whether Pyinstaller works correctly with them.

Once I create the .exe on each of them:
1. Windows opens.
2. And closes in the same second.

I'm on Python 3.10.6
All dependencies are in the virtual environment.
And the GUIs work when run as python app.py

I tried to resolve (or rather - test) with time.sleep() but the same issue continues.
I also tried --noconsole option for building. Still there.
(plus the search on this forum but no one seems to have the same issue)

Can you help please to make GUI windows stay on at least one of these GUI packages?

**Comment by rokm** [comment]
Try building without `--noconsole`, and run the .exe from command prompt (`cmd`) instead of double-clicking on it. What is the output in the console?

**Reply by EricTomaszewski** [reply]
Thanks.
That's a surprise => this is working.

When started from cmd through name.exe - the GUI window of both: Kivy & PyQt5 starts and displays the functional GUI.

But when double-clicking on the same .exe => the window appears and disappears in the same second.



As for the console output:

1. Absolutely none for PyQt.
2. For Kivy => similar to the one in Python terminal when running .py:

[INFO   ] [Logger      ] Record log in C:\Users\irene\.kivy\logs\kivy_23-11-05_0.txt
[INFO   ] [Kivy        ] v2.2.1
[INFO   ] [Kivy        ] Installed at "C:\Users\irene\PYTHON_CODE\LinkedIn\LinkedinPosting\dist\GUIFrame\_internal\kivy\__init__.pyc"
[INFO   ] [Python      ] v3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
[INFO   ] [Python      ] Interpreter at "C:\Users\irene\PYTHON_CODE\LinkedIn\LinkedinPosting\dist\GUIFrame\GUIFrame.exe"
[INFO   ] [Logger      ] Purge log fired. Processing...
[INFO   ] [Logger      ] Purge finished!
[INFO   ] [Factory     ] 190 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2 (img_pil, img_ffpyplayer ignored)
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] GLEW initialization succeeded
[INFO   ] [GL          ] Backend used <glew>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 - Build 30.0.101.1273'>
[INFO   ] [GL          ] OpenGL vendor <b'Intel'>
[INFO   ] [GL          ] OpenGL renderer <b'Intel(R) UHD Graphics'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 - Build 30.0.101.1273'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Text        ] Provider: sdl2
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Start application main loop
[INFO   ] [Base        ] Leaving application in progress...


**Does that give you any clue why this might be please and how to get it running through clicking on .exe [which is why I'm doing it => to be able to send exe to some non-techy person who doesn't even know what cmd is...]?**

**Reply by bwoodsend** [reply]
On Windows, it normally means you're referencing data files wrong. e.g. `open("file.txt")` instead of using `__file__`.

**Reply by EricTomaszewski** [reply]
Thanks for this Brenainn.

Yes. I'm on Windows 11.
I looked into the issue but can't find anything in my code that would reference any external files.

**Are you able to perhaps point please what could be corrected in this code?**

The script is a very simple Kivy GUI with input, button, output (presenting the input after pressing the button):
```
class LinkShareApp(App):

    def build(self):
        self.title = 'Link Share App'
        layout = BoxLayout(orientation='vertical', padding=10, spacing=10)
        
        self.link_input = TextInput(hint_text="Enter the link to share")
        self.output_label = Label(text="Output will be displayed here")

        share_button = Button(text="Share Link")
        share_button.bind(on_press=self.share_link)

        layout.add_widget(self.link_input)
        layout.add_widget(share_button)
        layout.add_widget(self.output_label)
        
        return layout

    def share_link(self, instance):
        link = self.link_input.text
        self.output_label.text = f"Sharing link: {link}"

if __name__ == '__main__':
    LinkShareApp().run()
```

**Reply by rokm** [reply]
What happens if you open `cmd`, and try to run your program from an arbitrary location? E.g., `cd c:\`, and then run the program with `c:\full\path\to\name.exe`? Does that work, or does it display error in the console?

**Reply by EricTomaszewski** [reply]
As in my first reply to rokm in this thread:
https://github.com/orgs/pyinstaller/discussions/8052#discussioncomment-7476875

It works correctly from the command line when running this .exe file.
No errors in the console.

But not when going to the windows explorer and running it by double clicking on .exe.

**Does that point you to anything please?**

**Reply by bwoodsend** [reply]
That's not what Rok was asking. He wants you to run from cmd but from a different location so that any relative path assumptions will raise errors that you can see.

**Comment by EricTomaszewski** [comment, marked as answer]
OK. Figured it out.
Python 3.12 _seems_ to be the problem 
(similar to what I found with packages like pyautogen & pymemgpt which do NOT install correctly on 3.12 => pyinstaller did pip install but then these errors disappeared when migrating back).

I migrated back to 3.11.2 and now everything works correctly with both: 
`pyinstaller app.py`
as well as
`pyinstaller app.py --onefile`

Thanks for your time guys.
Pyinstaller is an awesome package to distribute to non-techy clients!
(this topic can be closed now)

**Reply by Akashmsagar0707** [reply]
I am stuck with an unusual problem. I have a GUI code written in Python PyQt5. It has images, put inside an image folder and many python files. This GUI is working fine in Pycharm. I am able to generate a standalone executable file for this project using Pyinstaller. When i run the exe , no images are visible in GUI . i have really tried hard and searched a lot of internet for the same , but still i am not getting this concept of getting the images . please help mw for this ..
